File: java/13-test-driven-exercise/src/test/java/org/first/tutorial/ExerciseTest.java
package org.first.tutorial;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;
import java.util.List;

class ExerciseTest {

    // ----------------------------------------------------------------------------------
    // INSTRUCTION: Test Driven Development (TDD)
    // ----------------------------------------------------------------------------------
    // 1. Write a test case below that fails (because the code doesn't exist yet).
    // 2. Go to Exercise.java and write just enough code to make it pass.
    // 3. Repeat.
    //
    // Goal: Implement CommandParser.parse(script)
    // - Input: String script (e.g., "FORWARD 1\nWAIT\nFORWARD 2")
    // - Output: List<String> of commands.
    // ----------------------------------------------------------------------------------

    @Test
    void testParseSingleCommand() {
        // TODO: Uncomment and fix the test
        // Exercise.CommandParser parser = new Exercise.CommandParser();
        // List<String> result = parser.parse("FORWARD 1");
        // assertEquals(1, result.size());
        // assertEquals("FORWARD 1", result.get(0));
        fail("Uncomment the test above and implement the code!");
    }

    @Test
    void testParseMultipleLines() {
        // TODO: Test with "FORWARD 1\nWAIT"
        // Should return list of size 2.
    }
}

File: kotlin/15-inheritance-and-interfaces/src/test/kotlin/org/first/tutorial/ExerciseTest.kt
package org.first.tutorial

import org.first.tutorial.Exercise.MecanumDrive
import org.first.tutorial.Exercise.TankDrive
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class ExerciseTest {

    @Test
    fun tankDriveFormatsAndClamps() {
        val drive = TankDrive()
        val output = drive.drive(0.8, 0.5) // left=1.3 -> clamp to 1.0, right=0.3
        assertTrue(output.contains("Tank"))
        assertTrue(output.contains("left:1.0"))
        assertTrue(output.contains("right:0.3"))
    }

    @Test
    fun mecanumDriveFormats() {
        val drive = MecanumDrive()
        val output = drive.drive(0.6, -0.4) // fl=0.2 fr=1.0 after clamp
        assertTrue(output.contains("Mecanum"))
        assertTrue(output.contains("FL:0.2"))
        assertTrue(output.contains("FR:1.0"))
        assertTrue(output.contains("RR:1.0"))
    }
}
